pkg/update/update.go

```diff
@@ -65,6 +65,60 @@ func (c *updater) WithStatusWriter(writer statuswriter.StatusWriter) *updater {
 	return c
 }
 
+func (c *updater) PrepareForVersion(
+	ctx context.Context,
+	pkgName string,
+	pkgVersion string) (*UpdateTransaction, error) {
+	c.status.Start()
+	defer c.status.Stop()
+	c.status.SetStatus("Collecting installed package")
+
+	var packagesToUpdate []v1alpha1.Package
```

Comment:
We don't need this in `PrepareForVersion`. There is always only one package to update. So you don't need to create this slice, and you don't need to iterate over it in line 91. It's just one package always.